File: src/main/java/no/fintlabs/security/CurrentAuditor.java
package no.fintlabs.security;

public final class CurrentAuditor {
    private static final ThreadLocal<String> TL = new InheritableThreadLocal<>();
    private CurrentAuditor() {}
    public static void set(String v) { TL.set(v); }
    public static String get()       { return TL.get(); }
    public static void clear()       { TL.remove(); }
}

File: src/main/kotlin/no/novari/flyt/configuration/security/CurrentAuditor.kt
package no.novari.flyt.configuration.security

object CurrentAuditor {
    private val currentAuditor = InheritableThreadLocal<String>()

    @JvmStatic
    fun set(value: String?) = currentAuditor.set(value)

    @JvmStatic
    fun get(): String? = currentAuditor.get()

    @JvmStatic
    fun clear() = currentAuditor.remove()
}
